docker-compose.yml

```diff
@@ -0,0 +1,51 @@
+version: '2.2'
+
+volumes:
+  boot:
+  state:
+  data:
+
+services:
+  provision:
+    image: alpine:3.13
+    command:
+      - sh
+      - -c
+      - |
+        if ! [ -f /mnt/boot/config.json ]; then
+          cp /config.json /mnt/boot/config.json
+        else
+          echo "INFO: Reusing already existing config.json in docker volume."
+        fi
+    volumes:
+      - boot:/mnt/boot
+      - ${CONFIG_JSON:-./config.json}:/config.json
+  os:
+    build:
+      context: .
+      dockerfile: Dockerfile.template
+      args:
+        OS_VERSION: ${OS_VERSION:-2.68.1_rev1}
+        DEVICE_TYPE: ${DEVICE_TYPE:-intel-nuc}
```

Comment:
@robertgzr perhaps we should use generic x64 image instead? woud it matter here?

Reply:
I don't think so. that one seems slower with it's updates though: https://hub.docker.com/r/resin/resinos/tags?page=1&ordering=last_updated&name=genericx86-64

Reply:
It's also 100MB larger. @alexgg looing ahead, wouldn't this be able to use the core hostapp in the future, the one that doesn't include the bsp stuff...?